torch/_inductor/ir.py

```diff
@@ -3404,54 +3404,55 @@ def get_template_tiling(self):
         )
 
 
-# class InplaceBernoulliFallback(ExternKernel):
-
-#         (x,) = [t.codegen_reference() for t in self.inputs]
-#         wrapper.writeline(
-#             f"{self.kernel}({x}, {', '.join(map(repr, self.constant_args))})"
-#         )
-
-#     def should_allocate(self):
-#         return False
+class Wait(ExternKernelAlloc):
+    def __init__(
+        self,
+        layout,
+        inputs,
+        constant_args=(),
+    ):
+        super().__init__(layout, inputs, constant_args)
 
-#     def get_mutation_names(self):
-#         assert isinstance(self.layout, MutationLayout)
-#         return (self.layout.target.get_name(),)
+    def codegen(self, wrapper):
+        (work,) = self.constant_args
```

Comment:
Should this be a normal input (not a constant input)?  I worry there won't be a dependency edge in the scheduler that forces this to run after the allreduce.

You might also want to create fake dependencies to ensure ordering.